### PR TITLE
[Plugin] expose 'read only' configuration option in Splunk On-Call plugin

### PR DESCRIPTION
**Pull request by mdb**
This PR ~creates a `splunkOnCall.readOnly` configuration option~ adds a `readOnly` property to the `SplunkOnCallEntityCard` component to address issue #9634.

By default, the plugin continues to render as represented in the screenshot below, displaying a "Create incident" link, as well as acknowledge/resolve controls in association with each incident...

splunk_on_call_read_only_false

However, when `readOnly` is configured (`<SplunkOnCallEntityCard readOnly />`, the plugin does not render the "Create incident" link, nor the acknowledge/resolve controls in association with each incident as demonstrated in the following screenshot...

splunk_on_call_read_only_true

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [X] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/splunk-on-call/README.md [hunk 1]
```diff
@@ -74,6 +74,27 @@ proxy:
...
+**WARNING**: In current implementation, the Splunk OnCall plugin requires the `/splunk-on-call` proxy endpoint be exposed by the Backstage backend as an unprotected endpoint, in effect enabling Splunk OnCall API access using the configured `SPLUNK_ON_CALL_API_KEY` for any user or process with access to the `/splunk-on-call` Backstage backend endpoint. See below for further configuration options enabling protection of this endpoint.
+
+### Read Only mode
+
+The Splunk OnCall plugin also supports a "read only" mode if you wish to suppress the rendering of UI controls to trigger-resolve-acknowledge incidents.
+
+```yaml
+# enable readOnly mode
+splunkOnCall:
+  readOnly: true
```
Comment (mdb) [inline plugins/splunk-on-call/README.md:86]: ☝️ Is there a better name for the configuration option?

**Comment by freben** [comment]
Haven't done the full review yet, but a quick first thought:

This feels like a thing that could ultimately be controlled by the permissions framework. The plugin would ask, "is this user permitted to raise incidents (on this particular entity, even)". Until then - hm, could this be done instead in the form of an optional prop on the root card element, such as `readonly?: boolean`? Since the adopter needs to place this card explicitly in their views anyway, it's close at hand for them to add that prop straight into their view hierarchy, instead of going via config at all.

**Comment by mdb** [comment]
> This feels like a thing that could ultimately be controlled by the permissions framework.

@freben Thanks for the response. Where can I read more about the permissions framework?

>  Until then - hm, could this be done instead in the form of an optional prop on the root card element, such as `readonly?: boolean`?

That sounds reasonable! Am I correct to infer you prefer this because it eliminates the arguable excess of exposing a `splunkOnCall.readOnly` YAML config option?

**Comment by freben** [comment]
Regarding permissions - docs are being put in place right now; for now, the original RFC can be a good place to start! https://github.com/backstage/backstage/pull/7761 Note that the sections actually expand.

File: plugins/splunk-on-call/src/components/EntitySplunkOnCallCard.tsx [hunk 2]
```diff
@@ -124,7 +129,7 @@ export const EntitySplunkOnCallCard = () => {
     : undefined;
 
   const eventsRestEndpoint =
-    config.getOptionalString('splunkOnCall.eventsRestEndpoint') || null;
+    config.getOptionalString('splunkOnCall.eventsRestEndpoint') || true;
```
Comment (freben) [inline plugins/splunk-on-call/src/components/EntitySplunkOnCallCard.tsx:132]: this is unrelated right?

File: plugins/splunk-on-call/src/components/EntitySplunkOnCallCard.tsx [hunk 3]
```diff
@@ -109,7 +109,12 @@ const useStyles = makeStyles({
   },
 });
 
-export const EntitySplunkOnCallCard = () => {
+type Props = {
+  readOnly?: boolean;
+};
+
+export const EntitySplunkOnCallCard = ({ readOnly }: Props) => {
```
Comment (freben) [inline plugins/splunk-on-call/src/components/EntitySplunkOnCallCard.tsx:116]: for ultimate cleanliness, could

- `export` the props above,
- rename the props to match component,
- unpack the props inside the function,
- mark both with `/** public */`

this will make for a ✨ neat ✨ api report (and future generated API docs)!

so in the end,

```ts
/** @public */
export const EntitySplunkOnCallCard = (props: EntitySplunkOnCallCardProps) => {
  const { readOnly } = props;
```

File: plugins/splunk-on-call/src/components/Incident/IncidentListItem.tsx [hunk 4]
```diff
@@ -241,12 +243,16 @@ export const IncidentListItem = ({
...
-          />
+          {!readOnly ? (
+            <IncidentAction
+              currentPhase={incident.currentPhase || ''}
+              incidentId={incident.entityId}
+              resolveAction={handleResolveIncident}
+              acknowledgeAction={handleAcknowledgeIncident}
+            />
+          ) : (
+            ''
```
Comment (freben) [inline plugins/splunk-on-call/src/components/Incident/IncidentListItem.tsx:254]: should use `null` instead as the standard react "just render nothing" marker

**Review by freben — COMMENTED** [review]
almost there!

File: plugins/splunk-on-call/src/components/Incident/IncidentListItem.tsx (hunk 4 above)
Comment (freben) [reply plugins/splunk-on-call/src/components/Incident/IncidentListItem.tsx:254]: Oh and you can also do `{!readOnly && <IncidentAction ... />}`, even shorter (`false` also just leads to no output)

File: plugins/splunk-on-call/src/components/EntitySplunkOnCallCard.tsx (hunk 3 above)
Comment (mdb) [reply plugins/splunk-on-call/src/components/EntitySplunkOnCallCard.tsx]: @freben Good call! I believe I've made these changes.

File: plugins/splunk-on-call/src/components/Incident/IncidentListItem.tsx (hunk 4 above)
Comment (mdb) [reply plugins/splunk-on-call/src/components/Incident/IncidentListItem.tsx]: @freben Thanks for the tips; I believe I've correctly made these changes.

**Review by freben — APPROVED** [review]
Yay!
